src/DotNetWorker.Grpc/GrpcWorker.cs

```diff
@@ -228,6 +233,36 @@ internal static WorkerInitResponse WorkerInitRequestHandler(WorkerInitRequest re
             return response;
         }
 
+        internal static async Task<FunctionMetadataResponses> FunctionsMetadataRequestHandlerAsync(FunctionsMetadataRequest request)
+        {
+            var response = new FunctionMetadataResponses
+            {
+                Result = StatusResult.Success
+            };
+
+            try
+            {
+                var functionReader = new FunctionMetadataJsonReader(request.FunctionAppDirectory);
```

Comment:
Should we consider abstracting this behind an interface ( ex: `IFunctionMetadataProvider`) so that writing unit tests for this class might be easier (you can `IFunctionMetadataProvider.ReadMetadataAsync` method. When we switch to the source gen implementation of this interface in future, you do not need to make any major changes to UTs.

Reply:
Yeah I agree with this one. In general, whenever you find yourself calling `new` on something, you should think if it'd be better to do it via DI. In this case, it does seem useful to be able to replace it for testing this method itself and for future replacement.